src/msix/common/AppxManifestObject.cpp

```diff
@@ -269,20 +269,17 @@ namespace MSIX {
     {
         ThrowErrorIf(Error::InvalidParameter, (capabilities == nullptr), "bad pointer.");
 
-        // Parse Capability elements.
         APPX_CAPABILITIES appxCapabilities = static_cast<APPX_CAPABILITIES>(0);
-        XmlVisitor visitorCapabilities(static_cast<void*>(&appxCapabilities), [](void* c, const ComPtr<IXmlElement>& capabilitiesNode)->bool
+        auto capabilitiesNames = GetCapabilities(APPX_CAPABILITY_CLASS_GENERAL);
+        for (const auto& capability : capabilitiesNames)
         {
-            APPX_CAPABILITIES* capabilities = reinterpret_cast<APPX_CAPABILITIES*>(c);
-            auto name = capabilitiesNode->GetAttributeValue(XmlAttributeName::Name);
-            const auto& capabilityEntry = std::find(std::begin(capabilitiesList), std::end(capabilitiesList), name.c_str());
+            const auto& capabilityEntry = std::find(std::begin(capabilitiesList), std::end(capabilitiesList), capability.c_str());

```

Comment:
>std::begin [](start = 52, length = 10)

Not that you did this, but if you are going to use standalone begin, you shouldn't qualify it so ADL kicks in.  More important for templates, but a reasonable best practice.